### ISSUE TITLE
Support for 3 points (and more) leveling in Manual leveling 

### ISSUE THREAD

**Issue by tobashadow** [andrivet/ADVi3pp#98]
**NOTE**: The exchange is summarized below:

Here's a topic and good read on why 4 point leveling is wrong and why 3 point is better ( I fully agree after moving to it) -

https://drmrehorst.blogspot.com/2017/07/3-point-print-bed-leveling-vs-4-point.html

On a I3 Plus or V2 for that matter all you do is purchase that upgraded Y carriage which is a popular upgrade anyways and rotate the stock heated bed CCW to the left 90 degrees and the third hole that is already drilled in your bed lines up. Remove the two bolts and springs on the left side and move one to the third hole and level as given in that article and as long as you haven't bent your board alreadydoing 4 point leveling, it will stay flat from then on.

This is a common mod to the I3 Plus utilizing this plate -

https://cdn.shopify.com/s/files/1/0002/9445/3292/products/PYP4-3_110x110@2x.jpg?v=1519236101

You can see the hole position on the left hand side where the 3 point level for that side goes.

**Comment by andrivet** [comment]
Please give a minimum of information (such as reference, etc.)

**Comment by tobashadow** [comment]
This is a common mod to the I3 Plus utilizing this plate - 

https://cdn.shopify.com/s/files/1/0002/9445/3292/products/PYP4-3_110x110@2x.jpg?v=1519236101

You can see the hole position on the left hand side where the 3 point level for that side goes.

**Comment by andrivet** [comment]
Sorry, still not getting it. How it works?

**Comment by tobashadow** [comment]
Ok here's a topic and good read on why 4 point leveling is wrong and why 3 point is better ( I fully agree after moving to it) -

https://drmrehorst.blogspot.com/2017/07/3-point-print-bed-leveling-vs-4-point.html

On a I3 Plus or V2 for that matter all you do is purchase that upgraded Y carriage which is a popular upgrade anyways and rotate the stock heated bed CCW to the left 90 degrees and the third hole that is already drilled in your bed lines up. Remove the two bolts and springs on the left side and move one to the third hole and level as given in that article and as long as you haven't bent your board alreadydoing 4 point leveling, it will stay flat from then on.

**Comment by Martin-Dolan** [comment]
I concur..  Three point level is the way to go. 
I didn't rotate my build plate 90° as the writing on the build surface would be sideways and my OCD would kill me. 
I just drilled an extra hole in my hotbed middle left and also countersunk all the holes (WTF is with the raised screw heads they use?)

If you are unsure of 4point vs 3point level, just think of why camera tripods have 3 legs. No matter what you put them on, it's impossible to get them to wobble.  Now try 4 legs. Impossible for them to *not* wobble.

NOTE:  not everyone has the single screw side on the left.  It could be on any of the 4 sides.  So you might get requests to be able to set which side it's on.

**Comment by tobashadow** [comment]
I just realized that a way around this is to change the manual leveling spots on the screen from 5  like this currently

```
X      X
    X 
X      X
```

To

```
X  X  X
X  X  X
X  X  X
```

This would not only support 3 point leveling with the 3rd point in any position but also add extra check points for leveling even in 4 point leveling, You could leave the 1- 5 numbering as normal and maybe put something like a star for the new four points and just add a edit to the manual stating they are extra positions for alternative 3 point leveling or other usage

```
4  *  2
 *  5  *
1  *  3
```

**Comment by andrivet** [comment]
@tobashadow This is probably the way to go.

**Comment by andrivet** [comment]
It may look like this:

![manualleveling](https://user-images.githubusercontent.com/981049/42287124-73f49dee-7fb5-11e8-8fe0-f56a475b7bbc.png)

**Comment by andrivet** [comment]
9 points implemented in version 4.0
